Sources/Apollo/RequestChainNetworkTransport.swift

```diff
@@ -81,18 +81,37 @@ open class RequestChainNetworkTransport: NetworkTransport {
   open func constructRequest<Operation: GraphQLOperation>(
     for operation: Operation,
     cachePolicy: CachePolicy,
-    contextIdentifier: UUID? = nil) -> HTTPRequest<Operation> {
-    JSONRequest(operation: operation,
-                graphQLEndpoint: self.endpointURL,
-                contextIdentifier: contextIdentifier,
-                clientName: self.clientName,
-                clientVersion: self.clientVersion,
-                additionalHeaders: self.additionalHeaders,
-                cachePolicy: cachePolicy,
-                autoPersistQueries: self.autoPersistQueries,
-                useGETForQueries: self.useGETForQueries,
-                useGETForPersistedQueryRetry: self.useGETForPersistedQueryRetry,
-                requestBodyCreator: self.requestBodyCreator)
+    contextIdentifier: UUID? = nil
+  ) -> HTTPRequest<Operation> {
+    let request = JSONRequest(
+      operation: operation,
+      graphQLEndpoint: self.endpointURL,
+      contextIdentifier: contextIdentifier,
+      clientName: self.clientName,
+      clientVersion: self.clientVersion,
+      additionalHeaders: self.additionalHeaders,
+      cachePolicy: cachePolicy,
+      autoPersistQueries: self.autoPersistQueries,
+      useGETForQueries: self.useGETForQueries,
+      useGETForPersistedQueryRetry: self.useGETForPersistedQueryRetry,
+      requestBodyCreator: self.requestBodyCreator
+    )
+
+    if Operation.operationType == .subscription {
+      request.addHeader(
+        name: "Accept",
+        value: "multipart/mixed;boundary=\"graphql\";\(MultipartResponseSubscriptionParser.protocolSpec),application/json"
+      )
+    }
+
+    if Operation.hasDeferredFragments {
```

Comment:
Is there a chance that we will have a subscription that also has deferred fragments? This would override that protocol spec in that case.

Reply:
No, I don't think defer is supported in subscriptions. In that case we'd be indirectly relying on graphql-js to invalidate the operation.

Reply:
Is graphql-js already handling that case?

Reply:
I went looking for validation of my assumption and it was not as easy to find as I thought it was. I did find reference to not being allowed on a Subscription in the [spec edits PR](https://github.com/graphql/graphql-spec/pull/742/files#diff-607ee7e6b71821eecadde7d92451b978e8a75e23d596150950799dc5f8afa43eR1559) though.

Reply:
As long as it's definitely not allowed in the specification, I'm fine with merging this then. If there is a bug in the future, that is a bug in GraphQL-js not Apollo-iOS. :)

Reply:
I think we're fine. IMO defer on subscriptions is pointless.

Reply:
OK, more confirmation in [Apollo Client's httpLink](https://github.com/apollographql/apollo-client/blob/main/src/link/http/createHttpLink.ts#L149-L150).